### PR TITLE
Pull Request for Feature branch!

### PR DESCRIPTION
**Pull request by luciano-ilha**
Pull request for evaluation and merge feature branch with master. Made all HTML structure, css styles and javascript code for the logic and functionality.

### CODE REVIEW

**Review by Forison — COMMENTED** [review]
# ⁉️Invalid
## Hi @luciano-ilha 
#### You need to set up and enable linter checks on this repo, Your pull request should be made up of title and a short description of what happened on the branch. These are mandatory requirements without which your code can not be review. Kindly address the issue and submit a new review request. You will find all info about linter configs below [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/)
#### Happy coding
Addo Forison

**Comment by luciano-ilha** [comment]
> # ⁉️Invalid
> ## Hi @luciano-ilha
> #### You need to set up and enable linter checks on this repo, Your pull request should be made up of title and a short description of what happened on the branch. These are mandatory requirements without which your code can not be review. Kindly address the issue and submit a new review request. You will find all info about linter configs below [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/)
> #### Happy coding
> Addo Forison

Sorry....I have stopped Microverse for some time and now that I'm back I forgot some stuff, a little bit rusty that's all. Getting back on track. Thanks

File: README.md [hunk 1]
```diff
@@ -45,17 +41,19 @@ To make sure the linters' checks using Github Actions works properly, you should
...
 ## Show your support :muscle:
 
 Give a ⭐️ if you like this project!
 
 ## :thumbsup: Acknowledgements
 
 - Microverse  
-- ThOdinProject
\ No newline at end of file
+- ThOdinProject
```
Comment (MuriloRoque) [inline README.md:62]: - [x] **(OPTIONAL)** It would be nice if you put your app online and give us a Live Link :smile:

File: main.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  key: "506cbac2b3ba4e29e7a6542e49fa22bb",
+  base: "https://api.openweathermap.org/data/2.5/",
+};
+
+const searchBox = document.querySelector(".topbox-search");
+searchBox.addEventListener("keypress", setQuery);
+
+let bodyTag = document.querySelector("body");
+
+function setQuery(event) {
```
Comment (MuriloRoque) [inline main.js]: - [x] Please use ES6 syntax arrow functions for these ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/es6_requirement.png))

File: main.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  const queryResult = await fetch(
+    `${api.base}weather?q=${query}&units=metric&APPID=${api.key}`
+  );
+
+  const weatherResponse = await queryResult.json();
+
+  return weatherResponse;
+};
+
+function displayResults(weather) {
```
Comment (MuriloRoque) [inline main.js]: - [x] Please use ES6 syntax arrow functions for these ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/es6_requirement.png))

File: main.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    bodyTag.style.backgroundImage = "url('/assets/weather-scshot.jpg')";
+  }
+
+  let hilo = document.querySelector(".currentbox-hilo");
+  hilo.innerText = `${Math.round(weather.main.temp_min)}°c / ${Math.round(
+    weather.main.temp_max
+  )}°c`;
+}
+
+function dateBuilder(d) {
```
Comment (MuriloRoque) [inline main.js]: - [x] Please use ES6 syntax arrow functions for these ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/es6_requirement.png))

File: main.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+const getResults = async (query) => {
+  const queryResult = await fetch(
+    `${api.base}weather?q=${query}&units=metric&APPID=${api.key}`
+  );
+
+  const weatherResponse = await queryResult.json();
+
+  return weatherResponse;
+};
```
Comment (MuriloRoque) [inline main.js]: - [x] Please handle the error that could happen in this API call, you could use a try/catch block here ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png))

File: package.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "weatherapp",
+  "version": "1.0.0",
+  "description": "A weather app, which allows the user to search for a specific location and get it's temperature, hi-lo temperatures and present date, displayed back.",
+  "main": "main.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
```
Comment (MuriloRoque) [inline package.json]: - [x] Please add the necessary scripts to build your app here ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/scripts_requirement.png))

**Review by MuriloRoque — CHANGES_REQUESTED** [review]
## Status: Changes requested :red_circle: 

### Hello @luciano-ilha :wave: 

### You have done an amazing job on this project, it's almost complete, you did almost everything that was asked in the requirements, we should just fix some things before moving. Please check my comments below, alright? :smile: 

- [x] You should either build a toggle button to switch between Celsius and Fahrenheit or give the user an option to choose between them before submitting ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png)).
- [x] You should change the look of your page depending on the weather of the location the user is searching ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png)).
- [x] You should use Webpack in this project as we learned in the previous one ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/webpack_requirement.png)). And also that's why I recommended that you add the necessary scripts in your `package.json`.

Kindly make the requested changes and submit them for another review, please :pray: 

I will be on Slack 24/7 in case you have any questions about this review :handshake: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Ilha Carbonell** 
Good job with the improvements for this project :muscle:  :muscle:
It meets most of the requirements :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:

- [x] Appearance should change according to the weather of the city
- [x] Keep the application logic separated from DOM manipulation tasks
- [x] Necessary run commands should be in package.json
`
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  }
`
- [x] webpack-dev-server should be included in your package.json
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 
Nice work making the requested changes. The right scripts have been added. Well done!
Let's make some changes so we can merge this PR.
- [x] The background image should change depending on the weather condition. It's raining in my location but on checking the weather, I expected the background to rainy image but that was not the case. Please fix that.
- [x] Your application logic and the view is still entangled. The `utils.js` file should only contain `js` logic and no `DOM` logic. All DOM logic should live in the `index.js` file.


### Martin Afani
#### Slack: whiz25

**Comment by luciano-ilha** [comment]
OK if the 3 of you had run the project on your machine you would see that the background image IS changing with the result, if it is raining, a rain picture shows, if it is clear, a sunny day shows up (see image below)! So STOP asking me to change this. As for the utils.js and index.js I choose to put DOM manipulation in utils and call the functionality in index file cause it's the entry point for webpack, it imports utils file, which makes more sense to me! So I disagree with these changes, sorry!

Screen Shot 2021-04-02 at 10 22 58

**Review by krys2fa — CHANGES_REQUESTED** [review]
## Great job on this @luciano-ilha 👏 
Well done on implementing the changes suggested by the previous Code Reviewers as well. 👍 
### Status: Changes Required ♻️ 

- [x] [REQUIRED] You still have issues with the background of your app loading. This is seen in both local and live versions of your app. It is most likely as a result of you not referencing the path to your images correctly. Please see below a screenshot of the web console showing the errors.
Screen Shot 2021-04-02 at 7 14 53 PM

- [x] **NB**: Sorry this wasn't mentioned earlier but you have to make a PR from your `feature` branch to your `development` branch in order to follow the Gitflow practice. Please ensure to do so in your next project.

Please request another review after implementing the changes.

Happy Coding 🖥️  😃 

###### You can contact me on slack (Chris Amanor) for any clarification.

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @luciano-ilha :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**